TACC/Core-Portal
# Quick/support nested horz dl comp

## Overview: ##

- Show that **vertical** (**default**) `DefinitionLists` can be nested.
- Fix styles so that **horizontal** _and_ **vertical** `Definition Lists` can be nested.
- Fix styles so that **compact** _and_ **default** `Definition Lists` can be nested.

## PR Status: ##


## Related Jira tickets: ##

* [FP-318](https://jira.tacc.utexas.edu/browse/FP-318)

## Summary of Changes: ##

1. __New__: Add nested lists to UI Patterns `DescriptionList`
2. __New__: Improve specificity of `DescriptionList` styles to support mix-and-match direction and density.

## Testing Steps: ##
1. Load UI Patterns.
2. Scroll to `DescritpionList`.
3. See old lists are unchanged.
4. See new _nested_ lists!

## UI Photos:

Changed items are highlighted with scruffy pink outline (scroll down).

![Nested DL Component](https://user-images.githubusercontent.com/62723358/89333032-00615180-d65a-11ea-9849-55f95e016a10.png)

## Notes: ##

PR #99 needed to nest definition lists. It can do so without this PR. This PR just adds illustration of it (in UI Patterns) and support for mix-and-match attributes when nesting.
